backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopIncremental.scala

```diff
@@ -80,11 +88,21 @@ object BloopIncremental {
   )(implicit equivS: Equiv[Stamp]): Task[(Boolean, Analysis)] = {
     val setOfSources = sources.toSet
     val incremental = new BloopNameHashing(log, reporter, uniqueInputs, options, profiler.profileRun, tracer)
-    val initialChanges = incremental.detectInitialChanges(setOfSources, previous, current, lookup, output)
+    val initialChanges = incremental.detectInitialChanges(setOfSources, previous, current, lookup, converter, output)
+    def isJrt(path: Path) = path.getFileSystem.provider().getScheme == "jrt"
     val binaryChanges = new DependencyChanges {
-      val modifiedBinaries = initialChanges.binaryDeps.toArray
+      val modifiedLibraries = initialChanges.libraryDeps.toArray
+
+      val modifiedBinaries: Array[File] = modifiedLibraries
+        .map(converter.toPath(_))
+        .collect {
+          // jrt path is neither a jar nor a normal file
+          case path if !isJrt(path) =>
```

Comment:
What is jrt is modified? We are dropping that information here and this can lead to undercompilations

Reply:
I asked that on discord and Eugene is not even sure if jrt is handled correctly by zinc. Anyways, this a deprecated API and I have no idea if we can get file from it (I tried)